frontend/lib/search/search_page.dart

```diff
@@ -50,14 +68,22 @@ class _SearchPageState extends State<SearchPage> {
             ResultsLoader(
               searchText: searchFieldText,
               categoryIds: _selectedCategories.map((e) => e.id).toList(),
-              coordinates: currentCoordinatesInput,
+              coordinates: _sortingMode == SortingMode.byDistance ? _coordinates : null,
             )
           ],
         ),
         LocationButton(
-          setCoordinates: (position) =>
-              setState(() => _coordinates = CoordinatesInput(lat: position.latitude, lng: position.longitude)),
-        )
+          setCoordinates: (position) => setState(() {
+            _coordinates = CoordinatesInput(lat: position.latitude, lng: position.longitude);
+            _sortingMode = SortingMode.byDistance;
+          }),
+          setSortingMode: (sortingMode) {
+            if (sortingMode == SortingMode.alphabetically || _coordinates != null) {
+              setState(() => _sortingMode = sortingMode);
+            }
+          },
+          sortingMode: _sortingMode!,
```

Comment:
```suggestion
          sortingMode: _sortingMode,
```

Reply:
Somehow needed, otherwise I get an error. Not sure why, I thought dart would handle stuff like this automatically.

Reply:
yes its just changeable by setting default sortingMode. so i you keep your solution. you can also keep this